### ISSUE TITLE
Golem Lord Argelmach and General Angerforge fix

### ISSUE THREAD

**Issue by Metalica** [classicdb/database#890]
//There where bugs in DB, so i fixed and post here the patch which is working correctly. 

http://wow-pro.com/dungeon_and_raid_guides/4860_jame039s_blackrock_depths_guide_brd
"General Angerforge is your first 'real' fight in BRD. So prepare it well and clear everything in his room before you engage him. Getting adds while fighting him would probably mean total wipeout."

https://www.youtube.com/watch?v=0tiZQayS_G0 - Video showing of cleaning up the 3 dwarf group before enganging boss.

 Linked group of 3 man as shown on video to the boss. 
INSERT INTO `world`.`creature_linking_template` (`entry`, `map`, `master_entry`, `flag`, `search_range`) VALUES ('8898', '230', '9033', '1', '45');
INSERT INTO `world`.`creature_linking_template` (`entry`, `map`, `master_entry`, `flag`, `search_range`) VALUES ('8899', '230', '9033', '1', '45');

http://wow-pro.com/dungeon_and_raid_guides/4860_jame039s_blackrock_depths_guide_brd
"After killing the General, go back up the stairs and then across the room down a few stairs. You will enter a large room, full of little engineers (non elite) and a few Golems (Elites). Pull very carefully in this room, and don't charge, PULL, way back to somewhere safe. You need to clear the whole room, because Lord Argelmach will call for help and the whole golems + little engineers will aggro.

Once you have the whole room clear, prepare for Argelmach, which is not an easy fight."

Linking whole room to the boss. All 4 difirent npc's must agroo.
UPDATE `world`.`creature_linking_template` SET `entry`='8900', `map`='230', `master_entry`='8983', `flag`='1', `search_range`='0' WHERE (`entry`='8900') AND (`map`='230');
UPDATE `world`.`creature_linking_template` SET `entry`='8906', `map`='230', `master_entry`='8920', `flag`='1', `search_range`='0' WHERE (`entry`='8906') AND (`map`='230');
UPDATE `world`.`creature_linking_template` SET `entry`='8907', `map`='230', `master_entry`='8983', `flag`='1', `search_range`='0' WHERE (`entry`='8907') AND (`map`='230');
UPDATE `world`.`creature_linking_template` SET `entry`='8920', `map`='230', `master_entry`='8983', `flag`='1', `search_range`='0' WHERE (`entry`='8920') AND (`map`='230');

**Comment by Tobschinski** [comment]
Argelmach is already linked to the creatures in the Manufactory.

```
# entry, map, master_entry, flag, search_range
'8900', '230', '8983', '1', '100'
'8907', '230', '8983', '1', '0'
'8920', '230', '8983', '1', '100'
'8906', '230', '8920', '1', '12'
```

**Comment by Metalica** [comment]
All search ranges have to bet set to 0, so it works. Atm it does not work. Feel free to test.

**Comment by cala** [comment]
`range = 0` will aggro all golems with the same entry in the whole instance. This shouldn't be used unless we are 100% sure all of those are indeed within the Manufactury. Another solution would be to increase the range to ensure it is high enough for all golems in the Manufactury.

**Comment by Metalica** [comment]
Agreed, solution is pretty simple :)

**Comment by cala** [comment]
I tested the two boss fights as they are in the current state of the DB and they are working as they should:
- the Anvilrage Reservists in front of General Angerforge are linked to him: it is not possible to pull them separately
- engaging Argelmach aggros all the NPCs in the Manufactory: the two different kinds of golems, the Weapon Technicians and the Doomforge Arcanasmith. They all come.

I suggest that you upgrade your DB (if not already done): we did a massive rework of BRD last autumn and maybe you missed it. Or it is also possible that some default ranges (visibility, active or aggro...) were changed in your mangosd.conf config file.

Anyway, I'm closing this issue because it is working as intended.
